pylint/lint/pylinter.py

```diff
@@ -158,6 +158,12 @@ def _load_reporter_by_class(reporter_class: str) -> type:
         "bad-plugin-value",
         "Used when a bad value is used in 'load-plugins'.",
     ),
+    "E0014": (
+        "Incorrect setting encountered in top level configuration-section '%s' : '%s'",
+        "bad-configuration-option-value",
+        "Used when a top section value can't be parsed because it does not belong to the top section "
```

Comment:
See other comment, but based on the `isinstance()` test I feel like this message is only emitted when the type of the value is incorrect?

Reply:
At the top level of the configuration if we do not have a dict it's not a proper top section header. We actually want to change that so we don't have to guess what is the checker that contain this option, but it's a breaking change.